scripts/vehicle_node.py

```diff
@@ -0,0 +1,292 @@
+
+        #Fill array of size l x w with Gaussian Noise.
+        l = int(ceil(self.length/self.resolution))
+        w = int(ceil(self.width/self.resolution))
+        grid_map = (l,w)
+        gaussian_array = np.random.normal(self.mu, self.sigma, grid_map)
+
+        #Filter the array to smoothen the variation of the noise
+        gaussian_array = gaussian_filter(gaussian_array, self.sigma_filter)
+
+        return gaussian_array
+
+    def velocity_cmd_callback(self, data):
+        
+        """
+        Updates the most recent command velocity to a twist variable
+        """
+        with self.lock:
+            self.twist = data
+
+
+
+    def update_position(self, event):
+
+        """
+        Computes the pose of the vehicle in the terrian map. 
+        """
+
+        #Create a copy of the most recent stored twist data to perform calculations
+        with self.lock:
+            velocity_data = copy.deepcopy(self.twist)
+
+
+        #time elapsed since last update position call
+        if hasattr(event, 'last_real'):
+            if event.last_real is None:
+                time = rospy.Duration(0.05)
+            else:
+                time = event.current_real - event.last_real
+        
+        time = time.to_sec()
+
+        #Calculate angle turned in the given time using omega = theta/time
+        angle = velocity_data.angular.z*time
+
+        #Calculate distance travelled in the given time using linear velocity = arc distance/time
+        distance = velocity_data.linear.x*time
+
+        #Calculate yaw of the robot
+        self.vehicle_yaw += angle
+
+        #Calculate vehicle x, y, z position coordinates
+        # TODO recalculate the position based on traveling in a circular arc.
+        self.pose.position.x += (distance)*cos(self.vehicle_yaw)
+        self.pose.position.y += (distance)*sin(self.vehicle_yaw)
+
+        #Calculate z position using linear interpolation and create cloud array
+        
+        #1. Create range to be used in interpolation function
+        x = np.arange(0, self.gaussian_array.shape[0]*self.resolution, self.resolution)
+        y = np.arange(0, self.gaussian_array.shape[1]*self.resolution, self.resolution)
+
+        #2. Create cloud array to be converted to point cloud for vizualization
+        #TODO do this without for loops
+        cloud = []
+        for i in range(self.gaussian_array.shape[0]):
+            for j in range(self.gaussian_array.shape[1]):
+                innerlist = []
+                innerlist.append(x[i])
+                innerlist.append(y[j])
+                innerlist.append(self.gaussian_array[i][j])
+                cloud.append(innerlist)
+
+        #3. Create interpolation function based on the ranges and gaussian data
+        f = RectBivariateSpline(x,y, self.gaussian_array)
+
+        #4. Find z value for x and y coordinate of vehicle using interpolation function
+        self.pose.position.z = f(self.pose.position.x, self.pose.position.y)
```

Comment:
Looks like you're only using the vehicle center point to compute the height of the vehicle.  This is a good and necessary step to first prove out.
Add a TODO to use the entire vehicle footprint to compute the vehicle height.